src/tflite_cpu.cpp

```diff
@@ -692,12 +630,13 @@ class MLModelServiceTFLite : public vsdk::MLModelService,
         return MLModelService::make_tensor_view(tensor_data, tensor_size_t, std::move(shape));
     }
 
-    // The mutex and condition variable needed to track our state
-    // across concurrent reconfiguration and invocation.
+    // Accesss to the module state is serialized. All configuration
+    // state is held in the `state` type to make it easier to destroy
+    // the current state and replace it with a new one.
     std::mutex state_lock_;
-    std::condition_variable state_ready_;
-    std::shared_ptr<state> state_;
-    bool stopped_ = false;
+
+    // In C++17, this could be `std::optional`.
```

Comment:
Potentially, we could compile this module in C++17, if we wanted to.

Reply:
When would this optional value ever be null? It's set in the constructor, only modified when the mutex is locked, and set to something valid before the mutex is unlocked again. I wonder if we can just make it a raw `struct state` now.

...it's possible I'm being too shortsighted and we should keep it as-is to hedge our bets against it needing to actually become optional in the future, though I don't see the details for when that would be important.

...I guess the way you structured this PR, `stopped_` has become "whether `state_` is null." So, maybe leave as-is.